src/match_data_object.cpp

```diff
@@ -6,24 +6,36 @@ Value MatchDataObject::array(int start) {
     auto size = (size_t)(m_region->num_regs - start);
     auto array = new ArrayObject { size };
     for (int i = start; i < m_region->num_regs; i++) {
-        if (m_region->beg[i] == -1) {
-            array->push(NilObject::the());
-        } else {
-            const char *str = &m_string->c_str()[m_region->beg[i]];
-            size_t length = m_region->end[i] - m_region->beg[i];
-            array->push(new StringObject { str, length });
-        }
+        array->push(group(i));
```

Comment:
This was basically just duplicating what `group` already did, so I consolidated it here.